src/cpp/fastdds/topic/DDSSQLFilter/DDSFilterFactory.cpp

```diff
@@ -456,6 +456,42 @@ DDSFilterFactory::~DDSFilterFactory()
     pool.clear();
 }
 
+static std::shared_ptr<xtypes::TypeObject> get_complete_type_object(
+        const char* type_name,
+        const TopicDataType* data_type)
+{
+    ReturnCode_t ret;
+
+    // Try to get the complete TypeObject from the type name
+    std::shared_ptr<xtypes::TypeObjectPair> type_objects = std::make_shared<xtypes::TypeObjectPair>();
+    ret = DomainParticipantFactory::get_instance()->type_object_registry().get_type_objects(
+        type_name, *type_objects);
+    if (RETCODE_OK == ret)
+    {
+        return std::make_shared<xtypes::TypeObject>(type_objects->complete_type_object);
+    }
+
+    // If not found, try to get the complete TypeObject from the type identifier
+    xtypes::TypeObject complete_type_object;
+    if (xtypes::EK_MINIMAL == data_type->type_identifiers().type_identifier1()._d())
```

Comment:
I would rather explicitly compare against `EK_COMPLETE` both identifiers with an `if()- else if()`. I am thinking in the case we have two `EK_MINIMAL` or even a `EK_BOTH`. If it would not be possible in anyway, ignore this comment

Reply:
594ca6b